src/app/features/user-details/user-details.component.ts

```diff
@@ -0,0 +1,130 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {Form, FormArray, FormBuilder, FormGroup, FormsModule, ReactiveFormsModule, Validators} from "@angular/forms";
+import {Observable, Subject, takeUntil} from "rxjs";
+import {UserService} from "../../services/user.service";
+import {ActivatedRoute} from "@angular/router";
+import {UserModel} from "../../shared/models/user.model";
+import {InputValidatorComponent} from "../input-validator/input-validator.component";
+import {FormSubmitButtonComponent} from "../form-submit-button/form-submit-button.component";
+import {RoleService} from "../../services/role.service";
+import {RoleModel} from "../../shared/models/role.model";
+import {AsyncPipe} from "@angular/common";
+
+@Component({
+  selector: 'app-user-details',
+  standalone: true,
+  imports: [
+    InputValidatorComponent,
+    FormSubmitButtonComponent,
+    ReactiveFormsModule,
+    AsyncPipe,
+    FormsModule
+  ],
+  templateUrl: './user-details.component.html',
+  styleUrl: './user-details.component.scss'
+})
+export class UserDetailsComponent implements OnInit, OnDestroy {
+  currentUser: FormGroup = this.fb.group({
+    firstName: ['', [Validators.required]],
+    lastName: ['', [Validators.required]],
+    username: ['', [Validators.required]],
+    roles: [[], [Validators.required]]
+  });
+  user: UserModel = <UserModel>{};
+  private destroy: Subject<void> = new Subject();
+  errorMessage: string = '';
+  editMode: boolean = false;
+
+  constructor(
+    protected userService: UserService,
+    protected roleService: RoleService,
+    private fb: FormBuilder,
+    private route: ActivatedRoute
+  ) {
+  }
+
+  ngOnInit(): void {
+    const userId: string | null = this.route.snapshot.paramMap.get('id');
+    if (userId) {
+      const fetchedUser: Observable<UserModel> = this.userService.getUser(userId);
+      if (fetchedUser) {
```

Comment:
instead of doing na if statement here, you can directly subscribe to the getUser method